crates/libs/windows/src/core/heap.rs

```diff
@@ -18,3 +18,12 @@ pub unsafe fn heap_free(ptr: RawPtr) {
         HeapFree(heap, HEAP_NONE, ptr);
     }
 }
+
+pub fn heap_string<T: Copy + Default + Sized>(value: &[T]) -> *const T {
+    let buffer = heap_alloc((value.len() + 1) * std::mem::size_of::<T>()).expect("Could not allocate string") as *mut T;
```

Comment:
Nit: Since the `T` here is uninitialized this should probably be `*mut MaybeUninit<T>`. It's probably fine to leave for now though 🤷

Reply:
I really only want to use this function for `u8` and `u16`. Be nice if Rust had traits for stuff like that. C++ for example has [the is_unsigned trait](https://en.cppreference.com/w/cpp/types/is_unsigned).

Reply:
Because the pointer points to uninit memory, the next line (where you make the slice) is UB.

Reply:
Feel free to suggest a better implementation or shoot me a PR.

Reply:
Pull request opened here: https://github.com/microsoft/windows-rs/pull/1667

Reply:
Kenny that's an unfortunate way to respond to UB reports. I encourage you to consider them a serious bug even if the person who points it out doesn't provide a solution at the same time. There's several ways to deal with uninit data in Rust, and `rylev` had already mentioned one of them for you just above.

Reply:
I wasn't trying to minimize the issue. I just wasn't sure how to address it. Sorry for the confusion.